### PR TITLE
Adop 2859 update citzen create application class

### PR DESCRIPTION
**Pull request by NewmanJustice**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ADOP-2859

### What’s changed ### 
- Updated src/test/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/CitizenCreateApplicationTest.java.
- Added a new parameterized test (shouldFormatHyphenatedCaseRef) using @CsvSource.
- Added coverage for:
16-digit case IDs (standard path)
shorter IDs that require left zero-padding
edge numeric inputs (1 and 0)
Assertions verify formatted output remains in ####-####-####-#### structure via the existing aboutToSubmit callback path.
### Why this change ### 
- Strengthens confidence that hyphenatedCaseRef is consistently generated for both typical and edge-case IDs.
- Reduces regression risk in callback output consumed downstream.
### Impact ### 
Test update to reflect changes in CitizenCreateApplication
### Validation ### 
- New parameterized unit test added to existing suite in CitizenCreateApplicationTest.
 -Existing tests in the class remain intact and continue to cover DSS metadata and base case setup.
### What changed ### 
- Updated src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/event/CitizenCreateApplication.java:
- Added defensive validation in aboutToSubmit(...) to reject invalid input:
details == null
details.getData() == null
details.getId() == null
details.getId() <= 0
- Throws IllegalArgumentException when required case details are missing/invalid.
- Extracted case reference formatting into formatHyphenatedCaseRef(Long caseId) and used it when setting hyphenatedCaseRef.

### Why this change ### 
- Prevents callback execution with invalid/null case payloads, reducing risk of null-handling failures at runtime.
- Ensures case reference formatting remains consistent (####-####-####-####) across typical and edge-case IDs.
### Impact ### 
- Behavioural improvement in callback robustness for invalid inputs.
- No API contract changes.
- Test coverage increased for case reference formatting logic and edge cases.
### Validation ### 
- Unit tests in CitizenCreateApplicationTest cover expected formatting scenarios.
- Existing assertions for DSS metadata and default adoption type remain in place. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!groovy
-@Library("Infrastructure")
+@Library("Infrastructure@2.0.0")
```
Comment (hfilmore) [inline Jenkinsfile_CNP:2]: revert to be done together with other changes

**Review by hfilmore — APPROVED** [review]
